### PR TITLE
[alertmanager] - 3286 Enahance test connection template

### LINKED ISSUE #3286: Add enable: false to Alertmanager test-connection.yaml
### Is your feature request related to a problem ?

alertmanager test-connection pod is failing, even though the alertmanager is healthy.

### Describe the solution you'd like.

I would like to have ability to disable of Alertmanager test connection pod via values.yaml

test:
  enabled: true|false
    
Wrap alertmanager's
`test-connection.yaml` template with

{{- if .Values.test.enabled }}
apiVersion: v1
kind: Pod
metadata:
  name: "{{ include "alertmanager.fullname" . }}-test-connection"
  labels:
    {{- include "alertmanager.labels" . | nindent 4 }}
  annotations:
    "helm.sh/hook": test-success
spec:
  containers:
    - name: wget
      image: busybox
      command: ['wget']
      args: ['{{ include "alertmanager.fullname" . }}:{{ .Values.service.port }}']
  restartPolicy: Never
{{- end }}

or via option to annotations"helm.sh/hook": test-failed.

I'm using helm template piped to istioctl (injecting istio-proxy sidecard as well) for multiple charts (alertmanager is only one of them,) at one and I have no control over order of when the aleretmanager or the connection-test pods start The test connection pod always Errors out with connection refuse as the alertmanager pod is not up yet, thus make it useless for me.

### Describe alternatives you've considered.

Deleting pod manually after deployment, when failing.

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by jkutes**
#### What this PR does / why we need it
Ability to enable/disable Alertmanager test connection pod and pass annotations via values.yaml
This is needed to control the pod in more complicated multi chart deployment and istio sidecar.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #3286 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/alertmanager/Chart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ icon: https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/a
 sources:
   - https://github.com/prometheus/alertmanager
 type: application
-version: 0.29.0
+version: 0.29.1
```
Comment (monotek) [inline charts/alertmanager/Chart.yaml:9]: new feature should raise minor version

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/alertmanager/Chart.yaml (hunk 1 above)
Comment (monotek) [inline charts/alertmanager/Chart.yaml:9]:
```suggestion
version: 0.30.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by jkutes — COMMENTED** [review]
make sense
Naseem, please approve!

File: charts/alertmanager/values.yaml [hunk 2]
```diff
@@ -251,3 +251,9 @@ extraVolumes: []
 extraEnv: []
   # - name: FOO
   #   value: BAR
+
+testFramework:
+  enabled: true
```
Comment (monotek) [inline charts/alertmanager/values.yaml:256]: Should we enable that by deafult?
I would let it disabled.

```suggestion
  enabled: false
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by jkutes** [comment]
Gentlemen, if you would approve.
Thank you.

**Comment by jkutes** [comment]
How can get this approved?

**Review by monotek — APPROVED** [review]

**Pull request merged**